packages/datadog-plugin-aws-sdk/src/services/bedrockruntime.js

```diff
@@ -0,0 +1,236 @@
+'use strict'
+
+const BaseAwsSdkPlugin = require('../base')
+const log = require('../../../dd-trace/src/log')
+
+const AI21 = 'AI21'
+const AMAZON = 'AMAZON'
+const ANTHROPIC = 'ANTHROPIC'
+const COHERE = 'COHERE'
+const META = 'META'
+const STABILITY = 'STABILITY'
+const MISTRAL = 'MISTRAL'
+
+class BedrockRuntime extends BaseAwsSdkPlugin {
+  static get id () { return 'bedrock runtime' }
+
+  generateTags (params, operation, response) {
+    let tags = {}
+    let modelName = ''
+    let modelProvider = ''
+    const modelMeta = params.modelId.split('.')
+    if (modelMeta.length === 2) {
+      [modelProvider, modelName] = modelMeta
+      modelProvider = modelProvider.toUpperCase()
+    } else {
+      [, modelProvider, modelName] = modelMeta
+      modelProvider = modelProvider.toUpperCase()
+    }
+
+    const shouldSetChoiceIds = modelProvider === COHERE && !modelName.includes('embed')
+
+    const requestParams = extractRequestParams(params, modelProvider)
+    const textAndResponseReasons = extractTextAndResponseReason(response, modelProvider, modelName, shouldSetChoiceIds)
+
+    tags = buildTagsFromParams(requestParams, textAndResponseReasons, modelProvider, modelName, operation)
+
+    return tags
+  }
+}
+
+function extractRequestParams (params, provider) {
+  const requestBody = JSON.parse(params.body)
+  const modelId = params.modelId
+
+  if (provider === AI21) {
+    const temperature = requestBody.temperature || ''
+    const topP = requestBody.topP || ''
+    const maxTokens = requestBody.maxTokens || ''
+    const stopSequences = requestBody.stopSequences || []
+    let prompt = requestBody.prompt
+
+    if (modelId.includes('jamba')) {
+      for (const message of requestBody.messages) {
+        if (message.role === 'user') {
+          prompt = message.content // Return the content of the first user message
```

Comment:
are we looking for the first user message or the last in the set of messages? we'll grab the last user message here, as we don't `break` from the loop in this condition, but maybe that's what we want (last message is most recent), but the comment suggests otherwise. double checking because i think this is new logic here that isn't in the Python integration

Reply:
Ah I initially aimed to get the first user message. But after reading the docs more I realized that it's the whole conversation with the model and we're going to be interested in the most recent user prompt instead of the first one. I need to change the comment.

Reply:
After taking a look at the docs it appears that for A21 labs models there are 2 series the Jurassic and the Jamba types which have different input formats. In the python implementation we only support the Jurassic model arch type. So I added support here for the Jamba one as well.